x/stakeibc/keeper/msg_server_add_validators_test.go

```diff
@@ -0,0 +1,102 @@
+package keeper_test
+
+import (
+	sdkmath "cosmossdk.io/math"
+	_ "github.com/stretchr/testify/suite"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/Stride-Labs/stride/v7/x/stakeibc/types"
+	stakeibctypes "github.com/Stride-Labs/stride/v7/x/stakeibc/types"
+)
+
+type AddValidatorsTestCase struct {
+	hostZone           stakeibctypes.HostZone
+	validMsg           stakeibctypes.MsgAddValidators
+	expectedValidators []*stakeibctypes.Validator
+}
+
+func (s *KeeperTestSuite) SetupAddValidators() AddValidatorsTestCase {
+	hostZone := stakeibctypes.HostZone{
+		ChainId:    "GAIA",
+		Validators: []*stakeibctypes.Validator{},
+	}
+
+	validMsg := stakeibctypes.MsgAddValidators{
+		Creator:  "stride_ADMIN",
+		HostZone: "GAIA",
+		Validators: []*types.Validator{
+			{Name: "val1", Address: "stride_VAL1", Weight: 1},
+			{Name: "val2", Address: "stride_VAL2", Weight: 2},
+			{Name: "val3", Address: "stride_VAL3", Weight: 3},
+		},
+	}
+
+	expectedValidators := []*types.Validator{
+		{Name: "val1", Address: "stride_VAL1", Weight: 1, DelegationAmt: sdkmath.ZeroInt()},
+		{Name: "val2", Address: "stride_VAL2", Weight: 2, DelegationAmt: sdkmath.ZeroInt()},
+		{Name: "val3", Address: "stride_VAL3", Weight: 3, DelegationAmt: sdkmath.ZeroInt()},
+	}
+
+	s.App.StakeibcKeeper.SetHostZone(s.Ctx, hostZone)
+
+	return AddValidatorsTestCase{
+		hostZone:           hostZone,
+		validMsg:           validMsg,
+		expectedValidators: expectedValidators,
+	}
+}
+
+func (s *KeeperTestSuite) TestAddValidators_Successful() {
```

Comment:
Can we add a negative weight test?

Reply:
it's not possible because the weight is a uint64